### PR TITLE
ARQGRA-467 There won't be any attempt to take a Screenshot, when no Drone context is available

### PR DESCRIPTION
**Pull request by MatousJobanek**

### CODE REVIEW

File: extension/screenshooter/src/main/java/org/arquillian/extension/recorder/screenshooter/browser/impl/BrowserScreenshooter.java [hunk 1]
```diff
@@ -145,18 +146,18 @@ public Screenshot takeScreenshot(File screenshotToTake, ScreenshotType type) {
...
-    private WebDriver getTakingScreenshotsBrowser() {
-
-        GrapheneContext context = null;
+    private WebDriver getTakingScreenshotsBrowser(File screenshotToTake) {
+        WebDriver result = null;
 
         try {
-            context = GrapheneContext.getContextFor(Default.class);
+            GrapheneContext context = GrapheneContext.getContextFor(Default.class);
+            result = ((GrapheneProxyInstance) context.getWebDriver(TakesScreenshot.class)).unwrap();
```
Comment (michbeck100) [inline extension/screenshooter/src/main/java/org/arquillian/extension/recorder/screenshooter/browser/impl/BrowserScreenshooter.java]: The problem isn't the Graphene context, but that there's no WebDriver available. This happens if you mix in-container arquillian tests and Graphene tests in one project and use the same arquillian.xml.
If the arquillian.xml contains the following:

```
   <extension qualifier="screenshooter">
     ...
   </extension>
```
then the Screenshooter extension kicks in and tries to create a screenshot if the in-container test fails. But context.getWebDriver fails with 
"There is no context available for qualifier org.jboss.arquillian.drone.api.annotation.Default. Available contexts are []"

Comment (MatousJobanek) [inline extension/screenshooter/src/main/java/org/arquillian/extension/recorder/screenshooter/browser/impl/BrowserScreenshooter.java]: But the Graphene context "keeps" the WebDriver (proxy), and when the method GrapheneContext.getWebDriver(...) is called it tries to retrieve it. And this is the moment when the ISE ("There is no context available for qualifier org.jboss.arquillian.drone.api.annotation.Default. Available contexts are []") should be thrown (for the incontainer tests). 
I don't know why, but I forgot the annotation `@RunAsClient` in my test case. Here is the fix: d5479c6d3dec76fffea1b5810251c41a40409005
This test passes without failing. If you have different observation, please share the reproducer. Thanks

Comment (michbeck100) [inline extension/screenshooter/src/main/java/org/arquillian/extension/recorder/screenshooter/browser/impl/BrowserScreenshooter.java]: Ok, now i see that you catch the IllegalStateException. Although i don't like catching RuntimeExceptions like these, this works. Thanks for your changes!

Comment (MatousJobanek) [inline extension/screenshooter/src/main/java/org/arquillian/extension/recorder/screenshooter/browser/impl/BrowserScreenshooter.java]: Well, I like this catching neither. If you have some better solution your PR is more than welcome ;-)

Comment (michbeck100) [inline extension/screenshooter/src/main/java/org/arquillian/extension/recorder/screenshooter/browser/impl/BrowserScreenshooter.java]: No i don't :-)

**Comment by MatousJobanek** [comment]
Landed in https://github.com/arquillian/arquillian-graphene/commit/3c17108227576d5f68f9be49a66c639ed3075311

**Pull request closed without merging**